### ISSUE TITLE
open can of worms:  es6 support for Promise (and other) Polyfills?

### ISSUE THREAD

**Issue by gregglind** [asbjornenge/testdom#1]
There are bunch of new things in es6.

Maybe a testdomES6 function that adds more? 

```
document.Promise = require('es6-promise').Promise;
```

**Comment by asbjornenge** [comment]
Funny, I read the email subject for this issue earlier today, and thought; "Well, at least it needs to be behind some sort of flag" which is exactly what you did in your PR :smile: :+1: 

However, I'm not sure I want to force lots of additional dependencies (even dev-deps) on people that don't need them. What if I instead allow passing an additional object of globals, like so:

```
require('testdom')(<html>, {
    'Promise' : 'es6-promise'
})
```

Where the key is the name of the global and the value is the module. This way users can use the modules they like to polyfill the globals they like. Another important point is to not require these modules if the browser provides them. Hence not calling require directly on the object here, but letting testdom require the value/module if no DOM exists. If someone wants to polyfill both JSDOM and browsers they can just set the global themselves.

I think I would rather keep a collection of "working polyfills / globals" in the README or something instead of bloating the dependencies.

Thoughts?

Also, I should really look into updating the jsdom version (I will) !!

**Comment by gregglind** [comment]
That sounds like a fine solution, if it works :)  Boolean flags are an anti pattern for sure.  

Testdom is a wonderful solution to a tedious little problem.

Thanks!

**Comment by asbjornenge** [comment]
@gregglind glad you like it :smile: Just merged in my suggested solution (+ updated jsdom) and published v1.0.0 to npm :tada: Take it for a spin :rocket:
